packages/main/src/SplitButton.js

```diff
@@ -0,0 +1,354 @@
+ * <code>ui5-split-button</code> can be pressed by clicking or tapping it, or by certain keys, which changes
+ * the style to provide visual feedback to the user that it is pressed or hovered over with
+ * the mouse cursor. A disabled <code>ui5-split-button</code> appears inactive and any of the two buttons
+ * cannot be pressed.
+ * <br><br>
+ * There are separate events that are fired on activating of <code>ui5-split-button</code> parts:
+ * <ul>
+ * <li><code>click</code> for the first button (text button)</li>
+ * <li><code>arrow-click</code> for the second button (arrow button)</li>
+ * </ul>
+ *
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/SplitButton.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SplitButton
+ * @extends UI5Element
+ * @tagname ui5-split-button
+ * @public
+ */
+class SplitButton extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SplitButtonCss;
+	}
+
+	static get template() {
+		return SplitButtonTemplate;
+	}
+
+	onAfterRendering() {
+		this.textButton.icon = this.buttonIcon;
+	}
+
+	_onFocusOut(event) {
+		if (this.disabled || event.isMarked) {
+			return;
+		}
+
+		this._focusedState(false);
+	}
+
+	_onFocusIn(event) {
+		if (this.disabled || event.isMarked) {
+			return;
+		}
+
+		this._focusedState(true);
+	}
+
+	_onMouseDownArrow(event) {
+		this._focusedState(false);
+	}
+
+	_onMouseTouchDownText(event) {
+		this._focusedState(false);
+		this._textButtonActiveState(true);
+	}
+
+	_onMouseUpText() {
+		this._textButtonActiveState(false);
+	}
+
+	_onKeyDown(event) {
+		if (isTabNext(event) || isTabPrevious(event)) {
+			this._tabIndexUpdate = true;
+		} else if (isDown(event) || isUp(event)) {
+			this.arrowButton.active = true;
+			this._fireArrowClick();
+		} else if (isSpace(event) || isEnter(event)) {
+			this.arrowButton.active = false;
+			this._updateIcon = true;
+			this._textButtonActiveState(true);
+			if (isEnter(event)) {
+				event.preventDefault();
+				this._fireClick();
+			} else {
+				this._spacePressed = true;
+			}
+		}
+
+		if (this._spacePressed && (isEscape(event) || checkModifierKeys(event, false, false, true))) {
+			this._shiftOrEscapePressed = true;
+			this._textButtonActiveState(false);
+		}
+	}
+
+	_onKeyUp(event) {
+		if (isDown(event) || isUp(event)) {
+			this.arrowButton.active = false;
+		} else if (isSpace(event) || isEnter(event)) {
+			this._textButtonActiveState(false);
+			if (isSpace(event)) {
+				if (!this._shiftOrEscapePressed) {
+					event.preventDefault();
+					event.stopPropagation();
+					this._fireClick();
+				} else {
+					this._shiftOrEscapePressed = false;
+				}
+				this._spacePressed = false;
+			}
+		}
+	}
+
+	_focusedState(state) {
+		this._tabIndexUpdate = true;
+		this.focused = state;
+	}
+
+	_textButtonActiveState(state) {
+		this.textButton.active = state;
```

Comment:
The same as the previous comment, you should not manipulate the DOM directly. Instead, create 2 private metadata properties called f.e. `_textButtonActive` and `_textButtonIcon`, and bind them to the active property and text content of the ui5-button in the shadow root respectively. When something happens, as in this method, just set new values to the 2 properties. Since they are metadata properties, they will trigger invalidation, the template function will be executed, and their new values will be set.

Reply:
Done.